### ISSUE TITLE
Remove application->database redirection

### ISSUE THREAD

**Issue by Bananeweizen** [cgeo/cgeo#2211]
Currently all of our database methods need to be invoked at cgeoapplication instead, which simply redirects to an instance of cgData with the same method signature. I intend to remove all these methods from cgeoapplication and to directly invoke them at cgData instead.

Reasoning: Every single member/field in cgData is static (besides the CacheCache, but we can make that one static, too). That means it is a pure utility class, not an object with some inner state. And that makes perfect sense, as we don't add any state to the underlying database, we only need a lot of methods to more easily access that underlying database.
As all fields are static, we can also make all methods of cgData static (which Eclipse even recommends, if we just change 2 method signatures of the current implementation).
As we don't have any state in cgData and as all methods can be static, there is absolutely no good reason to have some kind of "entry point" in cgeoapplication to "shield" all the database methods from being accessed from anywhere in the application.

Now some of you might object that this destroys the nice design we currently have. It doesn't, really. Currently our database is only accessed from the application class. That's nice. But every method that wants something from the database has a direct dependency to the cgeoapplication class instead. By removing all the indirection methods in cgeoapplication, we change neither
- the dependency count between all of our classes
- the dependency kinds
- the cyclomatic complexity
- ...

Or to re-phrase it a third time, if you are still not convinced: Why is it OK to have hundreds of dependencies from all classes to the cgeoapplication class, but it is not OK to instead have the same number of dependencies directly to the database class?

Basically this change will remove some hundred lines of code without any functional impact, just the receivers of database related methods will change all over the place.
